### PR TITLE
Switch content selector to default tree-structure #4933

### PR DESCRIPTION
**Pull request by vbradnitski**

### CODE REVIEW

**Comment by alansemenov** [comment]
Fantastic job, Slava!

Small issues:
1. When I expand the content selector in the Create Issue dialog, the dropdown is supposed to expand above the selector, but as seen on the screenshot below it's expanded under and is not visible:

image

That's how it used to be:

image

2. Small styling issue - display content name on highlighted row must be in white, not gray:
image

3. Selected content items must follow the same order as they were in the selector. For example, in the screenshot above `Large tree node 2` was before `Large tree node 2-100`, and they should be listed in the same order in the list of selected options:
image

4. For some reason the order of items inside Content Selector in "All input types" content type is not the same as in the grid. But at the same time in the Content Selector in the "Create Issue" it's correct

Content Grid:
image

Content Selector in the Create Issue dialog (same order):
image

Content Selector in the "All input types" content type in Features app (wrong order):
image

**Comment by vbradnitski** [comment]
fixed.

**Comment by alansemenov** [comment]
Two more issues:
1. Selection (checked state of checkboxes) gets lost when collapsing/expanding nodes in the tree (unless items were already selected when the dropdown was opened)
2. In the "Create Issue" dialog order of selected items is the opposite to what it should be (correct in "All input types"). We should still put a new batch of selected items on top, but the order inside a batch must be the same as in the tree.

**Comment by alansemenov** [comment]
This is still wrong:

> In the "Create Issue" dialog order of selected items is the opposite to what it should be (correct in "All input types"). We should still put a new batch of selected items on top, but the order inside a batch must be the same as in the tree.

**Comment by aro** [comment]
@vbradnitski  The logic of the code in `ContentResource.treeSelectorQuery` is a bit hard to follow. What exactly is the filter doing with the paths?

I think the new method in `ContentPath.getParentPath` should be called something else. It does not return the parent if you pass something else than 1. I'm also unsure if this method should be in ContentPath, seems too specific to be in API, might not be used for anything else. At least rename to getAncestorPath, getSubPath(levels) ?.I would move it to `ContentResource` where it's used.

There are no calls to the new `ContentPaths` method, taking a Stream. And there are anyway enough variants already. This class should be left as it was.

Is there any change in what `NodeStorageServiceImpl.doReturnNodes` does? It's less readable, more verbose and probably less efficient since it now creates an intermediate list.
Same in `ContentNodeHelper`, it's less readable after the change and does the same. I think sometimes a for is better than a stream with a lambda expression.

**Comment by alansemenov** [comment]
Merged to "next" branch.
